tests/unit/CircuitBreaker.spec.ts

```diff
@@ -0,0 +1,283 @@
+        test('should init', () => {
+            expect(() => new CircuitBreaker({})).not.toThrow();
+        });
+
+        test('should always be in nominal state', () => {
+            b = new CircuitBreaker({});
+            b._evaluate();
+            expect(b.state).toStrictEqual(BreakerState.Nominal);
+        });
+    });
+
+    describe('missing configuration', () => {
+        test('should init', () => {
+            expect(() => new CircuitBreaker(undefined)).not.toThrow();
+        });
+
+        test('should always be in nominal state', () => {
+            b = new CircuitBreaker(undefined);
+            b._evaluate();
+            expect(b.state).toStrictEqual(BreakerState.Nominal);
+        });
+    });
+
+    test('should throw if config does not validate', () => {
+        const config = {
+            probes: 2,
+        };
+        expect(() => new CircuitBreaker(config)).toThrowErrorMatchingSnapshot();
+    });
+
+    describe('_evaluate', () => {
+        test('should set state to nominal if all probes evaluate to true', () => {
+            const config = {
+                probes: [
+                    {
+                        type: 'noop',
+                        returnConstantValue: true,
+                    },
+                ],
+            };
+            b = new CircuitBreaker(config);
+            b._evaluate();
+            expect(b.state).toStrictEqual(BreakerState.Nominal);
+        });
+
+        test('should set state to tripped if one probe evaluates to false', () => {
+            const config = {
+                probes: [
+                    {
+                        type: 'noop',
+                        returnConstantValue: false,
+                    },
+                    {
+                        type: 'noop',
+                        returnConstantValue: true,
+                    },
+                ],
+            };
+            b = new CircuitBreaker(config);
+            b._evaluate();
+            expect(b.state).toStrictEqual(BreakerState.Tripped);
+        });
+
+        test('should set state to stabilizing if all probes evaluate to true after being tripped', () => {
+            const config = {
+                probes: [
+                    {
+                        type: 'noop',
+                        returnConstantValue: true,
+                    },
+                ],
+            };
+            b = new CircuitBreaker(config);
+            b._aggregateState = BreakerState.Tripped;
+            b._evaluate();
+            expect(b.state).toStrictEqual(BreakerState.Stabilizing);
+        });
+
+        test('should set state to nominal if all probes evaluate to true N times while stabilizing', () => {
+            const config = {
+                probes: [
+                    {
+                        type: 'noop',
+                        returnConstantValue: true,
+                    },
+                ],
+                stabilizeAfterNSuccesses: 2,
```

Comment:
I get the reasoning that nominal == stabilized, but it's one step too far and doesn't help clarity because nominal loses context.

`stabilize<condition>` explicitly applies to the `stabilizing` case whereas `nominal` is kind of context-less. If we took it to the extreme, `nominalAfterNSuccesses` could also apply to the initial state where we would wait for N successes before allowing any activity to take place, which is wrong.

I spent quite a bit of time trying to get the naming unambiguous (although too long for my taste), so I'll leave it as is, unless you have an even more unambiguous suggestion :)

Reply:
to keep with the circuit breaker metaphor might I suggest open/closed

Reply:
That's where it gets confusing: circuits are open/closed but breakers are nominal/tripped (which results in the circuit open or closed).

Reply:
The only word I can think of now would be `SuccessesUntilStabilized` (or Stable/Nominal) but this is quite similar to your initial name, so please feel free to ignore this comment for now.